pkg/lexer/stateFunctions_test.go

```diff
@@ -34,72 +33,110 @@ func TestStateFunction_matchIdentifier(t *testing.T) {
 		//Test for matchIdentifier
 		genIdentifier := rapid.StringMatching(`[[:alnum:]]+`).Draw(t, "genIdentifier")
 		lIdentifier := NewFromString(genIdentifier)
-		//fmt.Printf("Generated string: %v \n", genIdentifier)
 		matchIdentifier(lIdentifier)
-		//fmt.Printf("lIdentifier: %v len: %v \n", lIdentifier.tokenList, len(lIdentifier.tokenList))
 		if len(lIdentifier.tokenList) == 0 {
 			t.Fatalf("TestStateFunction_matchIdentifier: Tokens were made, Tokenlist: %v ", lIdentifier.tokenList)
 		}
 	})
 }
 
-// !DO NOT WORK!
 func TestStateFunction_matchLetters(t *testing.T) {
 	rapid.Check(t, func(t *rapid.T) {
-		genLetters := rapid.StringMatching(`node hey {}`).Draw(t, "genLetters")
-		fmt.Printf("%v \n", genLetters)
+		// Test for letters using matchLetters
+		genLetters := rapid.StringMatching(`[[:alpha:]]+`).Draw(t, "genLetters")
 		lLetter := NewFromString(genLetters)
 		matchLetters(lLetter)
-
-		fmt.Printf("lLetter: %v len: %v \n", lLetter.tokenList, len(lLetter.tokenList))
-
-		if len(lLetter.tokenList) == 0 {
+		if len(lLetter.tokenList) > 0 {
 			t.Fatalf("TestStateFunction_matchLetters: Tokens were made, Tokenlist: %v ", lLetter.tokenList)
 		}
+		// Test for letters using matchLetters & matchIdentifier
+		matchIdentifier(lLetter)
+		if len(lLetter.tokenList) == 0 {
+			t.Fatalf("TestStateFunction_matchLetters_matchIdentifier: No tokens were made, Tokenlist: %v ", lLetter.tokenList)
+		}
+		// Test for letters using matchLetters & matchKeyword
+		matchKeyword(lLetter)
+		if len(lLetter.tokenList) == 0 {
+			t.Fatalf("TestStateFunction_matchLetters_matchKeyword: No tokens were made, Tokenlist: %v ", lLetter.tokenList)
```

Comment:
Can this be merged since it prints the same thing?

Reply:
be the change you want to see :))

Reply:
ahvafornoget

Reply:
Du kan prøve se hvad der sker